libvast/test/system/index.cpp

```diff
@@ -84,17 +85,19 @@ struct fixture : fixtures::deterministic_actor_system_and_events {
     ids result;
     size_t collected = 0;
     auto fetch = [&](size_t chunk) {
-      for (size_t i = 0; i < chunk; ++i)
-        self->receive(
-          [&](ids& sub_result) {
-            ++collected;
-            result |= sub_result;
-          },
-          after(0s) >> [&] {
-            FAIL("missing sub result #" << (i + 1) << " in partition #"
-                                        << (collected + 1));
-          }
-        );
+      auto done = false;
+      while (!done)
+        self->receive([&](ids& sub_result) { result |= sub_result; },
+                      [&](system::done_atom) { done = true; },
+                      caf::others >> [](caf::message_view& msg)
+                        -> caf::result<caf::message> {
+                        FAIL("unexpected message: " << msg.content());
+                        return caf::none;
+                      },
+                      after(0s) >> [&] { FAIL("ran out of messages"); });
```

Comment:
Would you mind explaining? The receive block is looks good to me. Everything is properly aligned and indented.

Reply:
I have to admit it would match our formatting guidelines. 😄 But the mix of indentation makes it hard to read. Perhaps it's just `caf::others` in the middle throws me off. I'm fine with leaving it as is.

Reply:
👍